### ISSUE TITLE
[powerpc64] e82ec2 breaks build

### ISSUE THREAD

**Issue by pkubaj** [rust-lang/rust#73324]
```
pkubaj@talos:$~$ rustc --version --verbose
rustc 1.44.0
binary: rustc
commit-hash: unknown
commit-date: unknown
host: powerpc64-unknown-freebsd
release: 1.44.0
LLVM version: 9.0
```

This is on FreeBSD head, ELFv2 on BE.

With commit, https://github.com/rust-lang/rust/commit/e82ec2315e5adb1c291c3702cd2ac1f46ecd0fcf applied, all Rust-using software fails to build.

The problem started with 1.43.0, built with 1.42.0 as a bootstrap. Attempting to build software with that build always failed with SIGSEGV. Rust 1.44.0 fails to build as well. Reverting this commit in 1.43.0 makes software build successfully, including Rust 1.44.0.

In 1.44.0, engine.rs was modified and moved to another location, but reverting the commit in question is still necessary. This the patch that we currently use:
https://svnweb.freebsd.org/ports/head/lang/rust/files/powerpc64-elfv1/patch-src_librustc__mir_dataflow_framework_engine.rs?view=markup

This is for ELFv1, but the patch for ELFv2 is just the same.

**Comment by cuviper** [comment]
I think this is the same issue I found in [rhbz1837660](https://bugzilla.redhat.com/show_bug.cgi?id=1837660), because the LLVM fix that I identified would apply to any big-endian target. It's possible that your issue may manifest in a different place -- I saw it in the code that processes full LTO, but the problem is not really LTO related at all.

It should be fixed by LLVM 10, which will hit Rust 1.45 (now in beta). If you're building with your own LLVM 9, or want to apply the fix to the bundled LLVM in Rust 1.43/44, you can use my backport:
https://src.fedoraproject.org/rpms/llvm/pull-request/49

**Comment by pkubaj** [comment]
Thanks, this indeed fixes build on both ELFv1 and ELFv2.
